agent/src/softwarecontaineragent.h

```diff
@@ -79,7 +80,7 @@ class SoftwareContainerAgentError : public SoftwareContainerError
 class SoftwareContainerAgent
 {
     LOG_DECLARE_CLASS_CONTEXT("SCA", "SoftwareContainerAgent");
-    typedef std::shared_ptr<SoftwareContainer> SoftwareContainerPtr;
+    typedef std::shared_ptr<ContainerAgentInterface> SoftwareContainerPtr;
```

Comment:
This name is super confusing. Why not SoftwareContainerInterface? It is an interface to SoftwareContainer, isn't it?